pilot/pkg/networking/core/v1alpha3/cluster_builder.go

```diff
@@ -928,14 +939,32 @@ func (cb *ClusterBuilder) applyUpstreamTLSSettings(opts *buildClusterOpts, tls *
 		if tls.Mode == networking.ClientTLSSettings_ISTIO_MUTUAL && mtlsCtxType == autoDetected {
 			transportSocket := c.cluster.TransportSocket
 			c.cluster.TransportSocket = nil
-			c.cluster.TransportSocketMatches = []*cluster.Cluster_TransportSocketMatch{
-				{
-					Name:            "tlsMode-" + model.IstioMutualTLSModeLabel,
-					Match:           istioMtlsTransportSocketMatch,
-					TransportSocket: transportSocket,
-				},
-				defaultTransportSocketMatch(),
+			if cb.hbone {
+				c.cluster.TransportSocketMatches = []*cluster.Cluster_TransportSocketMatch{
+					{
+						Name:            "hbone",
+						Match:           hboneTransportSocketMatch,
+						TransportSocket: InternalUpstreamSocket,
+					},
+					{
+						Name:            "tlsMode-" + model.IstioMutualTLSModeLabel,
+						Match:           istioMtlsTransportSocketMatch,
+						TransportSocket: transportSocket,
+					},
+					defaultTransportSocketMatch(),
+				}
+			} else {
+				c.cluster.TransportSocketMatches = []*cluster.Cluster_TransportSocketMatch{
+					{
+						Name:            "tlsMode-" + model.IstioMutualTLSModeLabel,
+						Match:           istioMtlsTransportSocketMatch,
+						TransportSocket: transportSocket,
+					},
+					defaultTransportSocketMatch(),
```

Comment:
Can define a var as  HboneOrPlaintextSocket

Reply:
No, since it depends on `transportSocket` which is dynamic